### ISSUE TITLE
End-to-End Dense Video Captioning with Masked Transformer

### ISSUE THREAD

**Issue by yskim0** [yskim0/One-Day-One-Paper#67]
![image](https://user-images.githubusercontent.com/48315997/129479480-f932f412-9d76-42fd-a2f3-76c6daaf32b8.png)

- dense video captioning → building two models
    - event proposal
    - captioning model
- components
    - encoder → encodes the video into representations
    <img width="368" alt="스크린샷 2021-08-15 오후 10 01 33" src="https://user-images.githubusercontent.com/48315997/129479468-29247350-2bae-4889-8291-e4fab30f0792.png">

    - proposal decoder → decodes from the encoding with different anchors to form video event proposals
        - based on ProcNet (sota on long dense event proposal)
    - captioning decoder → masking network to restric its attention to the proposal event over the encoding feature

- loss : 학습하는 방향
![image](https://user-images.githubusercontent.com/48315997/129479483-7c08a3ab-35e5-4675-8933-fd02ee4fb442.png)

The loss for training our model has four parts: the regres- sion loss Lr for event boundary prediction, the binary cross entropy mask prediction loss Lm, the event classification loss Le (i.e. prediction Pe), and the captioning model loss Lc. The final loss L is a combination of these four losses
